python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -478,6 +478,45 @@ def my_sensor(context):
 
         return materialization_by_partition
 
+    @public
+    def latest_materialization_records_per_asset_by_partition(
+        self,
+    ) -> Mapping[str, List[Tuple[AssetKey, "EventLogRecord"]]]:
```

Comment:
Thoughts on making this `-> Mapping[str, Mapping[AssetKey, EventLogRecord]]`? I think it's very rare that anyone would care about partition materializations before the latest one per partition. Sorry if I misled with my comment on the other PR.

Reply:
No problem, this makes sense, I've updated the method to have the new function signature